src/particles/particle_spin_pol.jl

```diff
@@ -177,3 +177,15 @@ y-polarized
 struct PolarizationY <: AbstractDefinitePolarization end
 const PolY = PolarizationY
 Base.show(io::IO, ::MIME"text/plain", ::PolY) = print(io, "y-polarized")
+
+"""
+    number_of_spin_pol(spin_or_pol)
+
+Return the number of spins or polarizations respresented by `spin_or_pol`, e.g. `number_of_spin_pol(SpinUp()) == 1`.
+
+"""
+function number_of_spin_pol end
+number_of_spin_pol(::AbstractDefinitePolarization) = 1
+number_of_spin_pol(::AbstractDefiniteSpin) = 1
+number_of_spin_pol(::AbstractIndefinitePolarization) = 2
+number_of_spin_pol(::AbstractIndefiniteSpin) = 2
```

Comment:
I'd suggest something like `spin_pol_multiplicity` maybe?

Reply:
In f5c8b35 I replaced `number_of_spin_pol` to just `multiplicity`. I think the addition of `spin_pol` to the name is redundant.